core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -92,7 +92,7 @@ public String generateFragment() {
         if(Jenkins.getInstance().hasPermission(Jenkins.READ)) {
             long now = System.currentTimeMillis();
             for (Downloadable d : Downloadable.all()) {
-                if(d.getDue()<now && d.lastAttempt+10*1000<now) {
+                if(d.getDue()<now && TimeUnit.SECONDS.toMillis(d.lastAttempt+10)<now) {
```

Comment:
In `old`, the modified part looked like 
```
(millis + millis) < now
```
while in `new` it's rather like 
```
toMillis(millis + seconds) < now
```
(as lastAttempt is set to `now` in millis at line 110).

Reply:
I agree with @pzygielo, the TimeUnit.SECONDS.toMillis should be used only with 10, not with d.lastAttempt+10 all together.

Reply:
thanks @pzygielo. I'll create a follow up ticket to fix this.